probe-rs/src/probe/mod.rs

```diff
@@ -208,6 +206,7 @@ impl Probe {
     /// Enters debug mode
     pub fn attach(mut self, target: impl Into<TargetSelector>) -> Result<Session, Error> {
         self.inner.attach()?;
+        self.attached = true;
```

Comment:
What this function should do if the probe is already attached?

Reply:
Then it just attaches again. This should be no issue. The issue with the double attach is that it might use a non available default protocol and crash.